test/src/test/java/org/corfudb/integration/CorfuReplicationMultiSenderIT.java

```diff
@@ -0,0 +1,588 @@
+        CountDownLatch statusUpdateLatch1 = new CountDownLatch(2);
+        corfuStoreSource1.openTable(LogReplicationMetadataManager.NAMESPACE,
+            LogReplicationMetadataManager.REPLICATION_STATUS_TABLE,
+            LogReplicationMetadata.ReplicationStatusKey.class,
+            LogReplicationMetadata.ReplicationStatusVal.class,
+            null,
+            TableOptions.fromProtoSchema(LogReplicationMetadata.ReplicationStatusVal.class));
+        ReplicationStatusListener statusListener1 =
+            new ReplicationStatusListener(statusUpdateLatch1);
+        corfuStoreSource1.subscribeListener(statusListener1,
+            LogReplicationMetadataManager.NAMESPACE, LR_STATUS_STREAM_TAG);
+
+        CountDownLatch statusUpdateLatch2 = new CountDownLatch(2);
+        corfuStoreSource2.openTable(LogReplicationMetadataManager.NAMESPACE,
+            LogReplicationMetadataManager.REPLICATION_STATUS_TABLE,
+            LogReplicationMetadata.ReplicationStatusKey.class,
+            LogReplicationMetadata.ReplicationStatusVal.class,
+            null,
+            TableOptions.fromProtoSchema(LogReplicationMetadata.ReplicationStatusVal.class));
+        ReplicationStatusListener statusListener2 =
+            new ReplicationStatusListener(statusUpdateLatch2);
+        corfuStoreSource2.subscribeListener(statusListener2,
+            LogReplicationMetadataManager.NAMESPACE, LR_STATUS_STREAM_TAG);
+
+        CountDownLatch statusUpdateLatch3 = new CountDownLatch(2);
+        corfuStoreSource3.openTable(LogReplicationMetadataManager.NAMESPACE,
+            LogReplicationMetadataManager.REPLICATION_STATUS_TABLE,
+            LogReplicationMetadata.ReplicationStatusKey.class,
+            LogReplicationMetadata.ReplicationStatusVal.class,
+            null,
+            TableOptions.fromProtoSchema(LogReplicationMetadata.ReplicationStatusVal.class));
+        ReplicationStatusListener statusListener3 =
+            new ReplicationStatusListener(statusUpdateLatch3);
+        corfuStoreSource3.subscribeListener(statusListener3,
+            LogReplicationMetadataManager.NAMESPACE, LR_STATUS_STREAM_TAG);
+
+        // Perform a role switch
+        Table<ExampleSchemas.ClusterUuidMsg, ExampleSchemas.ClusterUuidMsg, ExampleSchemas.ClusterUuidMsg> configTable =
+            corfuStoreSource1.openTable(DefaultClusterManager.CONFIG_NAMESPACE, DefaultClusterManager.CONFIG_TABLE_NAME,
+                ExampleSchemas.ClusterUuidMsg.class, ExampleSchemas.ClusterUuidMsg.class,
+                ExampleSchemas.ClusterUuidMsg.class, TableOptions.fromProtoSchema(ExampleSchemas.ClusterUuidMsg.class));
+
+        try (TxnContext txn = corfuStoreSource1.txn(
+            DefaultClusterManager.CONFIG_NAMESPACE)) {
+            txn.putRecord(configTable, DefaultClusterManager.OP_SWITCH, DefaultClusterManager.OP_SWITCH,
+                DefaultClusterManager.OP_SWITCH);
+            txn.commit();
```

Comment:
Same questions as before, in case of multiple-sources what does the switchover do? (include in the comments)... how are we validating that the data now arrived to all new sinks?